projects/rocprofiler-sdk/source/lib/python/rocpd/otf2.py

```diff
@@ -23,13 +23,437 @@
 
+import os
+import otf2
+from otf2.enums import LocationType, LocationGroupType, RegionRole, Paradigm
+import shutil
+import time
+from collections import defaultdict
+
 from .importer import RocpdImportData
 from . import output_config
-from . import libpyrocpd
+
+
+def get_perfetto_category_name(category):
+    """Map category names to perfetto category names"""
+    category_map = {
+        "NONE": "none",
+        "HSA_CORE_API": "hsa_api",
+        "HSA_AMD_EXT_API": "hsa_api",
+        "HSA_IMAGE_EXT_API": "hsa_api",
+        "HSA_FINALIZE_EXT_API": "hsa_api",
+        "HIP_RUNTIME_API": "hip_api",
+        "HIP_COMPILER_API": "hip_api",
+        "MARKER_CORE_API": "marker_api",
+        "MARKER_CORE_RANGE_API": "marker_api",
+        "MARKER_CONTROL_API": "marker_api",
+        "MARKER_NAME_API": "marker_api",
+        "MEMORY_COPY": "memory_copy",
+        "MEMORY_ALLOCATION": "memory_allocation",
+        "KERNEL_DISPATCH": "kernel_dispatch",
+        "SCRATCH_MEMORY": "scratch_memory",
+        "CORRELATION_ID_RETIREMENT": "none",
+        "RCCL_API": "rccl_api",
+        "OMPT": "openmp",
+        "RUNTIME_INITIALIZATION": "none",
+        "ROCDECODE_API": "rocdecode_api",
+        "ROCJPEG_API": "rocjpeg_api",
+        "HIP_STREAM": "hip_api",
+        "HIP_RUNTIME_API_EXT": "hip_api",
+        "HIP_COMPILER_API_EXT": "hip_api",
+        "ROCDECODE_API_EXT": "rocdecode_api",
+        "KFD_EVENT_PAGE_MIGRATE": "kfd_events",
+        "KFD_EVENT_PAGE_FAULT": "kfd_events",
+        "KFD_EVENT_QUEUE": "kfd_events",
+        "KFD_EVENT_UNMAP_FROM_GPU": "kfd_events",
+        "KFD_EVENT_DROPPED_EVENTS": "kfd_events",
+        "KFD_PAGE_MIGRATE": "kfd_events",
+        "KFD_PAGE_FAULT": "kfd_events",
+        "KFD_QUEUE": "kfd_events",
+    }
+    return category_map.get(category, "none")
+
+
+def allocation_level_type_name(level, type):
+    if level == "REAL":
+        name = "MEMORY"
+    elif level == "VIRTUAL":
+        name = "MEMORY_VMEM"
+    elif level == "SCRATCH":
+        name = "SCRATCH_MEMORY"
+    else:
+        return "UNKNOWN_LEVEL"
+
+    if type == "ALLOC":
+        return name + "_ALLOCATE"
+    elif type == "FREE":
+        return name + "_FREE"
+    else:
+        return level + "_MEMORY_NONE"
+
+    return name
 
 
 def write_otf2(importData, config):
-    return libpyrocpd.write_otf2(importData, config)
+
+    timer_resolution = 1_000_000_000
+    trace_dir = getattr(config, "output_path", "./otf_traces")
+    trace_file = f"{getattr(config, 'output_file', 'traces')}_results"
+    if not os.path.exists(trace_dir):
+        os.makedirs(trace_dir)
+
+    print(trace_dir)
+    if os.path.exists(trace_dir):
+        trace_subdir = os.path.join(trace_dir, trace_file)
+        if os.path.exists(trace_subdir):
+            shutil.rmtree(trace_subdir)
+        
+        otf2_file = os.path.join(trace_dir, f"{trace_file}.otf2")
+        def_file = os.path.join(trace_dir, f"{trace_file}.def")
+        
```

Comment:
```suggestion
        
        print(f"Writing: {otf2_file}")
        print(f"Writing: {def_file}")
        print(f"Writing: {trace_subdir} directory")
```
How about instead of the print above in line 105, we give the user some feedback like this @mcrnobrn-amd

Reply:
Done.